### PR TITLE
profile load fix

### PR DESCRIPTION
**Pull request by Len0u**
# Pull Request

## Description
Added a version-based cache invalidation system so profile pictures update instantly after upload or removal without refreshing the whole page.

## Type of Change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Related Issues (put task name here from Notion)
(Bug Fix) Profile Picture Load Fix

### Code Review Preparation
- [X] I have performed a self-review and reviewed the changed files
- [X] My code follows the project's style guidelines and passes linting
- [X] I have added comments in hard-to-understand areas and updated documentation

### Testing
- [ ] I have added relevant tests and they pass locally
- [X] I have tested the changes manually, including edge cases

### Code Quality & Security
- [X] I have formatted my code and removed debugging/unused code
- [X] No TypeScript errors or linting issues
- [X] No sensitive information committed (API keys, passwords, etc.)
- [X] User inputs are validated and sanitized

## Screenshots/Screen recording
https://github.com/user-attachments/assets/f4bf76a4-e9ee-4c54-b24d-a045c6f1f382

## Additional Notes
None.

## Summary by CodeRabbit

* **New Features**
  * Profile picture now updates instantly after upload or removal without requiring a page reload, providing immediate visual feedback and a faster, more responsive user experience.

### CODE REVIEW

File: src/pages/Profile/components/ProfilePicture/ProfilePicture.tsx [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
+  useEffect(() => {
+    setImageError(false);
+  }, [uid, version]);
+
   const defaultSrc =
     "https://i.pinimg.com/474x/33/f8/26/33f8266681c946cd80de486c499fe992.jpg";
 
   const src = uid
-    ? `https://storage.googleapis.com/for-all-ages-cdn/profile-pictures/${uid}?t=${Date.now()}`
+    ? `https://storage.googleapis.com/for-all-ages-cdn/profile-pictures/${uid}${version !== undefined ? `?t=${version}` : ""}`
```
Comment (evtran4) [inline src/pages/Profile/components/ProfilePicture/ProfilePicture.tsx:23]: smart fix!

**Review by evtran4 — APPROVED** [review]
